### ISSUE TITLE
cmd.run with multiple commands - random order of execution

### ISSUE THREAD

**Issue by kiemlicz** [saltstack/salt#42137]
### Description of Issue/Question
List of commands passed to `cmd.wait` (`cmd.run` also I guess) executes in **random** order
I expect it to be executed in order of appearance (especially that I've used _list_ to define them)
I don't want to use `cmd.script` or any other state that require having external file with commands/template for commands
Only pillar, only state definition.

### Setup
fancy.sls
```
any_fancy_id:
  cmd.run:
    - names: {{ some.commands.list.from.pillar }}    
```
somewhere in pillar:
```
some:
  commands:
    list:
      from:
        pillar:
         - "ls -al"
         - "ls -alR /"
         - "ls -alR /tmp"
         - "ls -alR /home"
```

### Steps to Reproduce Issue
`salt 'definately.wont.work.com' state.apply any_fancy_id`

### Versions Report
```
Salt Version:
           Salt: 2016.11.6
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: unknown
       dateutil: 2.4.2
      docker-py: 1.10.6
          gitdb: 0.6.4
      gitpython: 2.0.5
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: 0.25.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.25.1
         Python: 2.7.12+ (default, Sep 17 2016, 12:08:02)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.2.0
 
System Versions:
           dist: Ubuntu 16.10 yakkety
        machine: x86_64
        release: 4.11.8-no5kill
         system: Linux
        version: Ubuntu 16.10 yakkety
 

```

**Comment by garethgreenaway** [comment]
@kiemlicz Thanks for the report.  You are correct, the order that the names is processed is not done in order.  Is it possible for you to break out the calls to cmd.run for each command so that you're able to use requires and the ordering features that Salt offers?

**Comment by garethgreenaway** [comment]
@kiemlicz Looks like we had fixed this in the next release (2017.7) so I've added a PR to back port it for 2016.11.

**Comment by kiemlicz** [comment]
I don't know the exact number of commands upfront.
On the other hand I believe it would be possible to
write some jinja magic to expand these commands to separate states but it is kind of tedious.
Nevertheless thank you for information about fix and backporting.
